src/App.jsx

```diff
@@ -14,33 +15,86 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('alphabet');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleSortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('length');
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setGoods(goods);
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const isResetVisible =
+    JSON.stringify(goods) !== JSON.stringify(goodsFromServer);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? 'is-light' : ''}`}
+          onClick={handleSortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
```

Comment:
Well done on highlighting the active button. It's a good practice to provide visual feedback to the user.